Interfaces/API/Job.py

```diff
@@ -1005,19 +1010,28 @@ def _toJDL( self, xmlFile = '' ): #messy but need to account for xml file being
     for param in paramList:
       paramsDict[param.getName()] = {'type':param.getType(), 'value':param.getValue()}
 
-    scriptname = 'jobDescription.xml'
     arguments = []
-    if self.script:
-      if os.path.exists( self.script ):
-        scriptname = os.path.abspath( self.script )
-        self.log.verbose( 'Found script name %s' % scriptname )
+    scriptname = 'jobDescription.xml'
+
+    if jobDescriptionObject is None:
+      # if we are here it's because there's a real file, on disk, that is named 'jobDescription.xml'
+      if self.script:
```

Comment:
is it on purpose to have these two if statements separated ? So self.script is set, but the file does not exist, you don't do anything ?